plugins/inputs/sqlserver/sqlserver_test.go

```diff
@@ -224,6 +227,32 @@ func TestSqlServer_HealthMetric(t *testing.T) {
 	require.False(t, acc2.HasMeasurement(healthMetricName))
 }
 
+func TestSqlServer_Retry(t *testing.T) {
+	fakeServer := "localhost\\fakeinstance1;Database=fakedb1;Password=ABCabc01;"
+
+	var logOutputBuffer bytes.Buffer
+	log.SetOutput(&logOutputBuffer)
+
+	server := &SQLServer{
+		Servers:      []string{fakeServer},
+		IncludeQuery: []string{"DatabaseSize", "MemoryClerk"},
+		HealthMetric: true,
+		RetryCount:   2,
+		AuthMethod:   "connection_string",
+	}
+
+	var acc testutil.Accumulator
+	require.NoError(t, server.Start(&acc))
+	require.NoError(t, server.Gather(&acc))
+
+	logOutputString := logOutputBuffer.String()
+
+	assert.False(t, strings.Contains(logOutputString, "W! [inputs.sqlserver] Error gathering data, retrying 3 more time(s)"))
+	assert.True(t, strings.Contains(logOutputString, "W! [inputs.sqlserver] Error gathering data, retrying 2 more time(s)"))
+	assert.True(t, strings.Contains(logOutputString, "W! [inputs.sqlserver] Error gathering data, retrying 1 more time(s)"))
+	assert.False(t, strings.Contains(logOutputString, "W! [inputs.sqlserver] Error gathering data, retrying 0 more time(s)"))
```

Comment:
Instead of checking the log, how about starting a network server and count the connection attempts?